### PR TITLE
Spring Boot health check

### PR DESCRIPTION
**Pull request by gytis**

### CODE REVIEW

File: docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc [hunk 1]
```diff
@@ -1,63 +1,59 @@
...
 
-Once you have the {SpringBoot} booster deployed, you will have two applications running: `hello-service` and `name-service`. The `name-service` exposes a REST endpoint which returns a name as a String. The `hello-service` exposes a REST endpoint that returns a hello message as a String. When you call the REST endpoint in `hello-service`, the `hello-service` application makes a call to `name-service` to obtain a name String to use in the hello message String. 
+Once you have the {SpringBoot} booster deployed, you will have a service called  `{app-name}` running that exposes a two REST endpoints: `/api/greeting` which returns a name as a String and `/api/killme`, which forces the service to become unresponsive as means to simulate a failure.
 
-. Use `curl` to execute a `GET` request against the `hello-service`. You can also use a browser to do this.
+In the following we demonstrate the basic interactions on the command line that can be used verify the service availability and simulate a failure that causes the OpenShift self-healing capabilities to be trigger on the service. Alternatively you can use the service web interface to perform these steps (See #4).
```
Comment (rhoads-zach) [inline docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc]: Rewording to use active voice and for clarity:

> The following steps demonstrate how to verify the service availability and simulate a failure. This failure of an available service causes the OpenShift self-healing capabilities to be trigger on the service. 
> \+
> NOTE: The below steps use the command line to interact with the service.  Alternatively, you can use the web interface to perform the same steps (See #4).

File: docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc [hunk 2]
```diff
@@ -1,63 +1,59 @@
...
+[source,options="nowrap",subs="attributes"]
 ----
-$ curl http://hello-service.192.168.99.100.nip.io/api/greeting
-Hello World!
+$ curl http://{app-name}-myproject.192.168.64.3.nip.io/api/greeting
+{"content":"Hello, World!"}
 ----
 
-. Scale down the number of `name-service` pods, which are used by `hello-service`, to 0.
+. To simulate a service internal failure and to trigger the OpenShift selfhealing capabilities, invoke the `/api/killme` endpoint and verify the availability of the `/api/greeting` endpoint shortly after that. It should return a HTTP status 503:
```
Comment (rhoads-zach) [inline docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc]: You want your steps to be a clear action thats easily readable. Ideally, this means starting each step with an obvious verb. Use additional text within the step for more details. Also, you need to make your steps consistent with their endings (colons or periods). If you are doing sentences, each step should end in a period.

>  . Invoke the `/api/killme` endpoint and verify the availability of the `/api/greeting` endpoint shortly after that. 
> \+
> Invoking the `/api/killme` endpoint simulates an internal service failure and triggers the OpenShift self-healing capabilities. When invoking `/api/greeting` after simulating the failure, the service should return a HTTP status 503.

File: docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc [hunk 3]
```diff
@@ -1,63 +1,59 @@
...
+$ curl http://{app-name}-myproject.192.168.64.3.nip.io/api/greeting
+
+<html>
+  <head><title>Error</title></head>
+  <body>503 - Service Unavailable</body>
+</html>
 ----
 
-. Use `curl` to execute a `GET` request against the `hello-service` again. This should return an error since `name-service`, which is used by `hello-service` is unavailable and has failed . You can also use a browser to do this.
+. While performing these operations you can use the OpenShift console, or the `oc` client tools to watch the self-healing capabilities in action. Use `oc get pods -w` to continuously watch the service state from another terminal window. You should see that number pods in a `READY` state moves to zero (`0/1`) and after a while (~30 sec to 1 min) back up to one (`1/1`). In addition the `RESTARTS` count increases every time you kill the service:
```
Comment (rhoads-zach) [inline docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc]: Same feedback as previous. Simple, active step, use a period, etc

>. Use `oc get pods -w` to continuously watch the self-healing capabilities in action.
> \+
> While invoking the service failure, you can watch the self-healing capabilities in action on OpenShift console, or with the `oc` client tools. You should see the number pods in a `READY` state move to zero (`0/1`) and after a short period (less than one minute) move back up to one (`1/1`). In addition the `RESTARTS` count increases every time you you invoke the service failure.

File: docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc [hunk 4]
```diff
@@ -1,63 +1,59 @@
...
-</body></html>
+$ oc get pods -w
+NAME                           READY     STATUS    RESTARTS   AGE
+{app-name}-1-26iy7   0/1       Running   5          18m
+{app-name}-1-26iy7   1/1       Running   5         19m
+
 ----
 
-. Scale up the number of `name-service` pods to 1.
+. Alternatively to the interaction using the terminal window, you can use the web interface provided by the service to invoke the different methods and watch the service move through the life cycle phases:
```
Comment (rhoads-zach) [inline docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc]: Same as previous feedback

> . _Optional_: Use the web interface to invoke the service.
> \+
> Alternatively to the interaction using the terminal window, you can use the web interface provided by the service to invoke the different methods and watch the service move through the life cycle phases.

File: docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc [hunk 5]
```diff
@@ -1,63 +1,59 @@
...
-$ curl http://hello-service.192.168.99.100.nip.io/api/greeting
-Hello World!
-----
\ No newline at end of file
+To run the integration tests, execute the following command:
+
+[source,bash,option="nowrap"]
+--
+mvn clean verify -Popenshift,openshift-it
+--
```
Comment (rhoads-zach) [inline docs/topics/health-check-mission-basic-interaction-spring-boot-tomcat.adoc]: For consistency, start your commands executed from a shell with a `$`:

```
 [source,bash,option="nowrap"]
 --
 $ mvn clean verify -Popenshift,openshift-it
 --
```

**Comment by gytis** [comment]
@rhoads-zach updated.

**Pull request closed without merging**